src/blob.rs

```diff
@@ -103,24 +105,19 @@ impl<'a> BlobObject<'a> {
             {
                 Ok(file) => return Ok((name, file)),
                 Err(err) => {
-                    if attempt == max_attempt {
+                    if attempt >= MAX_ATTEMPT {
                         return Err(BlobError::CreateFailure {
                             blobdir: dir.to_path_buf(),
                             blobname: name,
                             cause: err,
                         });
                     } else {
                         name = format!("{}-{}{}", stem, rand::random::<u32>(), ext);
+                        continue;
```

Comment:
Actually surprising that this doesn't create any merge conflicts with https://github.com/deltachat/deltachat-core-rust/pull/2978/... Probably still makes sense to rebase and then quickly look at the diff again whether things look as expected.

For example: The `attempt == 1 && !dir.exists().await {` branch won't have a `continue`. It should either be in both branches, or in none. My favorite would be to just remove it.

Reply:
still looks fine after rebasing

Reply:
and yes, agree on the `continue`, removed it